strings/anagram/isanagram.go

```diff
@@ -0,0 +1,41 @@
+// isanagram.go
+// description: Checks if two given strings are anagram or not
+// details: An anagram is a word or phrase that's formed by rearranging the letters of another word or phrase
+// Reference: https://www.geeksforgeeks.org/pangram-checking/
+// Author : Venkatasubramanian
+
+package anagram
+
+import (
+	"regexp"
+	"strings"
+)
+
+func cleanString(text string) string {
+	cleanText := strings.ToLower(text)                      // Convert to lowercase
+	cleanText = strings.Join(strings.Fields(cleanText), "") // Remove spaces
+	regex, _ := regexp.Compile(`[^\p{L}\p{N} ]+`)           // Regular expression for alphanumeric only characters
```

Comment:
The current implementation without the `cleanString()` should work, since the `range` keyword automatically breaks the string into valid utf-8 runes. Removal of `cleanString()` and changing the `[]int` to a `map[rune]int` should work. Also, anagrams exclude whitespace, so you can either still include a different `cleanString()` implementation, or run a check in the range loop with `unicode.IsSpace()`.

Reply:
Sure will resolve it within EOD